nemo/collections/common/parts/preprocessing/collections.py

```diff
@@ -489,3 +481,253 @@ def _parse_item(self, line: str, manifest_file: str) -> Dict[str, Any]:
+
+        super().__init__(data)
+
+
+class DiarizationSpeechLabel(DiarizationLabel):
+    """`DiarizationLabel` diarization data sample collector from structured json files."""
+
+    def __init__(
+        self,
+        manifests_files: Union[str, List[str]],
+        emb_dict: Dict,
+        clus_label_dict: Dict,
+        round_digit=2,
+        seq_eval_mode=False,
+        pairwise_infer=False,
+        *args,
+        **kwargs,
+    ):
+        """
+        Parse lists of audio files, durations, RTTM (Diarization annotation) files. Since diarization model infers only
+        two speakers, speaker pairs are generated from the total number of speakers in the session.
+
+        Args:
+            manifest_filepath (str):
+                 Path to input manifest json files.
+            emb_dict (Dict):
+                Dictionary containing cluster-average embeddings and speaker mapping information.
+            clus_label_dict (Dict):
+                Segment-level speaker labels from clustering results.
+            round_digit (int):
+                Number of digits to be rounded.
+            seq_eval_mode (bool):
+                If True, F1 score will be calculated for each speaker pair as in the validation accuracy in training mode.
+            pairwise_infer (bool):
+                If True, this Dataset class operates in inference mode. In inference mode, a set of speakers in the input audio
+                is split into multiple pairs of speakers and speaker tuples (e.g. 3 speakers: [(0,1), (1,2), (0,2)]) and then
+                fed into the diarization system to merge the individual results.
+            *args: Args to pass to `SpeechLabel` constructor.
+            **kwargs: Kwargs to pass to `SpeechLabel` constructor.
+        """
+        self.round_digit = round_digit
+        self.emb_dict = emb_dict
+        self.clus_label_dict = clus_label_dict
+        self.seq_eval_mode = seq_eval_mode
+        self.pairwise_infer = pairwise_infer
+        audio_files, durations, rttm_files, offsets, target_spks_list, sess_spk_dicts, clus_spk_list, rttm_spk_list = (
+            [],
+            [],
+            [],
+            [],
+            [],
+            [],
+            [],
+            [],
+        )
+
+        for item in manifest.item_iter(manifests_files, parse_func=self.__parse_item_rttm):
+            if self.pairwise_infer:
+                if item['rttm_filepath']:
+                    base_scale_index = max(self.emb_dict.keys())
+                    _sess_spk_dict = self.emb_dict[base_scale_index][item['uniq_id']]['mapping']
+                    sess_spk_dict = {int(v.split('_')[-1]): k for k, v in _sess_spk_dict.items()}
+                    rttm_speaker_digits = [int(v.split('_')[1]) for k, v in _sess_spk_dict.items()]
+                else:
+                    sess_spk_dict = None
+                    rttm_speaker_digits = None
+
+                if self.seq_eval_mode:
```

Comment:
move this as discussed to if case of `item[rttm_filepath]`

Reply:
moved

Reply:
relocated the if statement